draft-johani-dnsop-multi-provider-requirements-00.md

```diff
@@ -148,6 +148,199 @@ to be able to fully support all multi-provider scenarios:
+DNSKEY RRset and adds KSK_An. Provider A monitors the propagation of the
+new DNSKEY RRset.
+
+When the DNSKEY RRset has fully propagated, provider A tries to agree with
+provider B on new CDS/CDNSKEY RRsets that include KSK_An (and no longer KSK_A).
+When agreement has been reached, provider A
+publishes the CDS/CDNSKEY RRset and assumes that provider B will do so as
+well.
+
+Provider A checks that the parent has a DS record for KSK_An. Then
+provider A tries to reach agreement with provider B to remove the CDS/CDNSKEY
+RRset.
+
+(Alternatively: both provider A and B monitor the DS RRset set the parent.
+When the DS RRset matches the agreed upon set of keys, each provider
+independently removes to CDS and CDNSKEY RRsets. Is there a risk?)
+
+### CSK rolls
+
+RFC 6781 recognizes Straightforward Rollover in a Single-Type Signing Scheme
+and Double-DS Rollover in a Single-Type Signing Scheme
+
+The Straightforward Rollover combines the elements of the Double-Signature
+ZSK roll with those of the Double-Signature KSK roll. No new requirements
+should be necessary.
+
+The Double-DS Rollover combines the Pre-Publish ZSK roll with the
+Double-DS KSK roll. Again no new requirements should be necessary.
+
+### Algorithm rolls
+
+RFC 6781 defines the conservative roll and somewhat describes the liberal
+roll. The conservative roll combines the Double-Signature ZSK roll with
+the Double-Signature KSK roll where the steps in the ZSK and KSK rolls are
+separated. No new requirements are expected.
+
+### Updating the NS RRset at the parent
+
+When provider A wants to change its NS RRset to NS_An, it can just update the
+NS RRset at the apex of the zone locally. Provider A then tries to agree with
+provider B on a new NS RRset. When agreement is reached provider A adds a
+CSYNC record to the zone.
+
+Provider A monitors the parent NS RRset. When provider A notices the parent
+NS RRset contains NS_An it can try to agree with provider B to remove the
+CSYNC record.
+
+### On-boarding a new provider
+
+To on-board provider C, provider MUST first ensure that its ZSK_C is in the
+DNSKEY RRsets of provider A and B. Then it MUST coordinate with provider A
+and C to update the DS RRset at the parent to refer to KSK_C. Finally
+provider C MUST coordinate with provider A and B to update the NS RRsets
+at the apex of A and B and at the parent to include NS_C.
+
+### Off-boarding 
+
+Off-boarding goes in the opposite order of on-boarding.
+
+### Requirements
+
+1) The zone owner specifies that provider A and provider B both sign
+   the zone and serve the zone. That A and B take full control over the 
+   NS RRset and that parent zone scans for CDS/CDNSKEY/CSYNC.
+
+2) The zone owner provides both providers with an unsigned copy of the zone.
+   The zone SHOULD NOT include the apex CDS, CDNSKEY, CSYNC, DNSKEY, and
+   NS RRsets. 
+
+   Open question: does the SOA version have any meaning beyond transferring 
+   the zone to the providers?
```

Comment:
Yes, its last field contains the TTL for negative caching. However, it's my view that this is not a reason to expose the SOA to the zone owner at all; I think it's fine for DNS operators to just hide it in all interfaces (like NSEC3PARAM). (That's actually what we do at deSEC.)